packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-node-ciemss.vue

```diff
@@ -39,43 +31,100 @@
+import { nodeMetadata } from '@/components/workflow/util';
 import { logger } from '@/utils/logger';
 import { Poller, PollerState } from '@/api/api';
 import type { WorkflowNode } from '@/types/workflow';
 import type { CsvAsset } from '@/types/Types';
-import type { RunResults } from '@/types/SimulateConfig';
 import { createLLMSummary } from '@/services/summary-service';
+import { createForecastChart } from '@/services/charts';
+import VegaChart from '@/components/widgets/VegaChart.vue';
 import type { CalibrationOperationStateCiemss } from './calibrate-operation';
 
 const props = defineProps<{
 	node: WorkflowNode<CalibrationOperationStateCiemss>;
 }>();
 const emit = defineEmits(['open-drilldown', 'update-state', 'append-output']);
 
-const modelConfigId = computed<string | undefined>(() => props.node.inputs[0]?.value?.[0]);
+const modelConfigId = computed<string | undefined>(() => props.node.inputs[0].value?.[0]);
 
-const runResults = ref<RunResults>({});
+const runResult = ref<any>(null);
 const csvAsset = shallowRef<CsvAsset | undefined>(undefined);
 
 const areInputsFilled = computed(() => props.node.inputs[0].value && props.node.inputs[1].value);
 const inProgressCalibrationId = computed(() => props.node.state.inProgressCalibrationId);
 
-const chartProxy = chartActionsProxy(props.node, (state: CalibrationOperationStateCiemss) => {
-	emit('update-state', state);
+let pyciemssMap: Record<string, string> = {};
+const preparedCharts = computed(() => {
+	const state = props.node.state;
+
+	if (!runResult.value || !csvAsset.value) return [];
+
+	const result = runResult.value;
+
+	const reverseMap: Record<string, string> = {};
+	Object.keys(pyciemssMap).forEach((key) => {
+		reverseMap[`${pyciemssMap[key]}`] = key;
+	});
+
+	// FIXME: Hacky re-parse CSV with correct data types
+	let groundTruth: Record<string, any>[] = [];
+	const csv = csvAsset.value.csv;
+	const csvRaw = csv.map((d) => d.join(',')).join('\n');
+	groundTruth = csvParse(csvRaw, autoType);
+
+	// Need to get the dataset's time field
+	const datasetTimeField = state.mapping.find(
+		(d) => d.modelVariable === 'timestamp'
+	)?.datasetVariable;
+
+	return state.chartConfigs.map((config) => {
+		const datasetVariables: string[] = [];
+		config.forEach((variableName) => {
+			const mapObj = state.mapping.find((d) => d.modelVariable === variableName);
+			if (mapObj) {
+				datasetVariables.push(mapObj.datasetVariable);
+			}
+		});
+
+		return createForecastChart(
+			{
+				dataset: result,
+				variables: config.map((d) => pyciemssMap[d]),
+				timeField: 'timepoint_id',
+				groupField: 'sample_id'
+			},
+			null,
+			{
+				dataset: groundTruth,
+				variables: datasetVariables,
+				timeField: datasetTimeField as string,
+				groupField: 'sample_id'
+			},
+			{
+				width: 180,
+				height: 120,
+				legend: false,
+				translationMap: reverseMap,
+				xAxisTitle: '',
+				yAxisTitle: ''
+			}
+		);
+	});
```

Comment:
There are actually some slight differences in terms of how the translation map is constructed, because the node does not need a legend while the drilldown does. 

But good point - I am collecting notes and data, I think once we do a cleanup pass over the optimization charts we will have enough to start refactor things.